pkg/deploy/registry/instantiate/rest.go

```diff
@@ -0,0 +1,286 @@
+package instantiate
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/registry/generic/registry"
+	"k8s.io/kubernetes/pkg/runtime"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/openshift/origin/pkg/client"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/deploy/api/validation"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+func NewREST(store registry.Store, oc client.Interface, kc kclient.Interface, codec runtime.Codec) *REST {
+	store.UpdateStrategy = Strategy
+	return &REST{store: &store, isn: oc, rn: kc, codec: codec}
+}
+
+// REST implements the Updater interface.
+var _ = rest.Updater(&REST{})
+
+type REST struct {
+	store *registry.Store
+	isn   client.ImageStreamsNamespacer
+	rn    kclient.ReplicationControllersNamespacer
+	codec runtime.Codec
+}
+
+func (s *REST) New() runtime.Object {
+	return &deployapi.DeploymentRequest{}
+}
+
+// Update instantiates a deployment config
+func (r *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
```

Comment:
Already covered why this was an update instead of a create?  Seems to me like I'm creating an instance, but if its already settled I won't push it.